library/include/rocwmma/internal/permute.hpp

```diff
@@ -39,12 +39,28 @@ namespace rocwmma
         struct Driver
         {
         private:
-            template <typename DataT, uint32_t VecSize, uint32_t... Idx>
-            ROCWMMA_DEVICE static inline auto
-                forEach(VecT<DataT, VecSize> const& src, uint32_t laneId, detail::SeqT<Idx...>)
+            template <typename DataT,
+                      std::enable_if_t<sizeof(DataT) == sizeof(uint64_t), bool> = true>
+            ROCWMMA_DEVICE static inline auto permute(DataT&& v, uint32_t laneId)
             {
-                static_assert(sizeof...(Idx) == VecSize, "Index count must match vector size");
-                return VecT<DataT, VecSize>{PermuteOp::exec(get<Idx>(src), laneId)...};
+                constexpr uint32_t B32VecSize = 2;
+                using B32VecT                 = VecT<uint32_t, B32VecSize>;
+
+                auto op = [](auto&& idx, auto&& v, uint32_t laneId) {
+                    constexpr auto i = std::decay_t<decltype(idx)>::value;
```

Comment:
Same comment as DPP

Reply:
also make sure not to introduce std:: usage

Reply:
fixed in commit - " Use vector_generator to replace forEach"